tiatoolbox/tools/tissuemask.py

```diff
@@ -0,0 +1,289 @@
+                shape (N, height, width, channels).
+        """
+        super().transform(images)
+
+        masks = []
+        for image in images:
+            grey = image[..., 0]
+            if len(image.shape) == 3 and image.shape[-1] == 3:
+                grey = cv2.cvtColor(image, cv2.COLOR_RGB2GRAY)
+            mask = (grey < self.threshold).astype(bool)
+            masks.append(mask)
+
+        return [mask]
+
+
+class MorphologicalMasker(OtsuTissueMasker):
+    """Tissue masker which uses a threshold and simple morphological operations.
+
+    This method applies Otsu's threshold before a simple small region
+    removal, followed by a morphological dilation. The kernel for the
+    dilation is an ellipse of radius 64/mpp unless a value is given for
+    kernel_size. MPP is estimated from objective power
+    via func:`tiatoolbox.utils.misc.objective_power2mpp` if a power
+    argument is given instead of mpp to the initialiser.
+
+    For small region removal, the minimum area size defaults to the area
+    of the kernel. If no mpp, objective power, or kernel_size areguments
+    are given then the kernel defualts to a size of 1x1.
+
+    The scale of the morphological operations can also be manually
+    specified with the `kernel_size` argument, for example if the
+    automatic scale from mpp or objective power is too large or small.
+
+    Examples:
+        >>> from tiatoolbox.tools.tissuemask import MorphologicalMasker
+        >>> from tiatoolbox.wsicore.wsireader import get_wsireader
+        >>> wsi = get_wsireader("slide.svs")
+        >>> thumbnail = wsi.get_thumbnail(32, "mpp")
+        >>> masker = MorphologicalMasker(mpp=32)
+        >>> masks = masker.fit_transform([thumbnail])
+
+        An example reading a thumbnail from a file where the objective
+        power is known:
+        >>> from tiatoolbox.tools.tissuemask import MorphologicalMasker
+        >>> from tiatoolbox.utils.misc import imread
+        >>> thumbnail = imread("thumbnail.png")
+        >>> masker = MorphologicalMasker(power=1.25)
+        >>> masks = masker.fit_transform([thumbnail])
+
+    """
+
+    def __init__(
+        self, *, mpp=None, power=None, kernel_size=None, min_region_size=None
+    ) -> None:
+        """Initialise a morphological masker.
+
+        Args:
+            mpp (float or tuple(float)):
+                The microns per-pixel of the image to be masked. Used to
+                calculate kernel_size a 64/mpp, optional.
+            power (float or tuple(float)):
+                The objective power of the image to be masked. Used to
+                calculate kernel_size as 64/objective_power2mpp(power),
+                optional.
+            kernel_size (int or tuple(int)):
+                Size of elliptical kernel in x and y, optional.
+            min_region_size (int):
+                Minimum region size in pixels to consider as foreground.
+                Defaults to area of the kernel.
+        """
```

Comment:
Please add a new line at the end.

Reply:
Fixed in this commit https://github.com/TIA-Lab/tiatoolbox/pull/61/commits/5b0e0ecc1fcb9ce989d907483bef9a2b0b2c7aa5